samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/ExportPoolListener.java

```diff
@@ -0,0 +1,101 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ **********************************************************************************/
+
+package org.sakaiproject.tool.assessment.ui.listener.author;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.faces.application.FacesMessage;
+import javax.faces.context.FacesContext;
+import javax.faces.event.AbortProcessingException;
+import javax.faces.event.ActionEvent;
+import javax.faces.event.ActionListener;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.commons.lang3.StringUtils;
+import org.sakaiproject.tool.assessment.facade.AgentFacade;
+import org.sakaiproject.tool.assessment.facade.AssessmentFacade;
+import org.sakaiproject.tool.assessment.facade.ItemFacade;
+import org.sakaiproject.tool.assessment.facade.QuestionPoolFacade;
+import org.sakaiproject.tool.assessment.services.QuestionPoolService;
+import org.sakaiproject.tool.assessment.services.assessment.AssessmentService;
+import org.sakaiproject.tool.assessment.ui.bean.authz.AuthorizationBean;
+import org.sakaiproject.tool.assessment.ui.bean.qti.XMLController;
+import org.sakaiproject.tool.assessment.ui.bean.qti.XMLDisplay;
+import org.sakaiproject.tool.assessment.ui.listener.util.ContextUtil;
+
+/**
+ * <p>Title: Samigo</p>
+ * <p>Description: Sakai Assessment Manager</p>
+ * @author jesusmmp
+ * @version $Id$
+ */
+@Slf4j
+public class ExportPoolListener implements ActionListener {
+
+	public void processAction(ActionEvent ae) throws AbortProcessingException {
+
+		String questionPoolId = (String) ContextUtil.lookupParam("questionPoolId");
+		String currentItemIdsString = (String) ContextUtil.lookupParam("currentItemIdsString");
+		XMLDisplay xmlDisp = (XMLDisplay) ContextUtil.lookupBean("xml");
+		String agentIdString = AgentFacade.getAgentString();
+		boolean accessDenied = true;
+		log.info("ExportPoolListener questionPoolId={}", questionPoolId);
+
+		QuestionPoolService questionPoolService = new QuestionPoolService();
+		QuestionPoolFacade questionPool = questionPoolService.getPool(Long.parseLong(questionPoolId),
+				AgentFacade.getAgentString());
+
+		List items = questionPoolService.getAllItems(questionPool.getQuestionPoolId());
+
+		StringBuilder sb = new StringBuilder();
+		// creating a question list separated by comma
+		for (Object item : items) {
+			if (sb.length() > 0) {
+				sb.append(",");
+			}
+			sb.append(((ItemFacade) item).getItemId());
+		}
+
+		// checking user can export pool
+		accessDenied = !questionPoolService.canExportPool(questionPoolId, agentIdString);
+
+		if (accessDenied) {
+			xmlDisp.setOutcome("exportDenied");
+			String thisIp = ((javax.servlet.http.HttpServletRequest) FacesContext.getCurrentInstance()
+					.getExternalContext().getRequest()).getRemoteAddr();
+			// logging IP , as requested in SAK-17984
+			log.warn("Unauthorized attempt to access /samigo-app/jsf/qti/exportAssessment.xml?questionPoolId="
```

Comment:
log.warn("Unauthorized attempt to access /samigo-app/jsf/qti/exportAssessment.xml?questionPoolId= {} from IP:{}", questionPoolId , thisIp);